### PR TITLE
Fix SplitPane movement

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**

### CODE REVIEW

**Comment by lpatiny** [comment]
It seems to me that this does not fix the problem (at least not in the preview).

What we expect as behaviour is something like 



You click on the splitter and you move the mouse. The mouse can leave the splitter but when the splitter moves the mouse is expected to be on it. In the change you did if the mouse leaves the parent div the control is lost and this is not what we expect neither. Once we 'mousedown' on the splitter we keep the control on it even if we quit the parent div for as long as we keep the mousedown (only exception is if we leave the browser window).

**Comment by wadjih-bencheikh18** [comment]
It wasn't clear in the issues
Aaaaaa okey now i understand the problem thank you

File: package.json [hunk 1]
```diff
@@ -40,16 +40,21 @@
...
     "@storybook/addon-actions": "^6.3.12",
     "@storybook/addon-essentials": "^6.3.12",
     "@storybook/addon-links": "^6.3.12",
     "@storybook/addon-storysource": "^6.3.12",
     "@storybook/react": "^6.3.12",
+    "@testing-library/dom": "^8.11.1",
+    "@testing-library/jest-dom": "^5.15.0",
+    "@testing-library/react": "^12.1.2",
+    "@testing-library/user-event": "^13.5.0",
+    "@types/jest": "^27.0.2",
```
Comment (targos) [inline package.json:54]: Please remove these dependencies. They were here in an older version of the project but are not used anymore.
After this is done, make sure you have npm 7 or 8, then delete the package-lock.json, run `npm install`, and commit the new package-lock.json.

Comment (wadjih-bencheikh18) [reply package.json:54]: i deleted those lines and run npm install => i got  `55 vulnerabilities (3 low, 15 moderate, 35 high, 2 critical)`
my npm version is 8.1.0

Comment (wadjih-bencheikh18) [reply package.json]: and i updated all packages before install (so this is why package-lock) changed

**Comment by wadjih-bencheikh18** [comment]
@lpatiny the issus is fixed now

**Comment by lpatiny** [comment]
It is much better indeed. There are still some small shifts and I wonder if it is easy to fix. Otherwise we can keep it like that 


.

**Comment by stropitek** [comment]
I can confirm that there are shifts. If you start jiggling your mouse fast around where the the split ends, you'll have large shifts.

**Comment by wadjih-bencheikh18** [comment]
This shifts is due to math.floor i think 
I can try fix it

**Comment by wadjih-bencheikh18** [comment]
@lpatiny it's better now

**Comment by wadjih-bencheikh18** [comment]
@lpatiny it works perfectly now
can you check please?

**Comment by wadjih-bencheikh18** [comment]
If the orientation is horizontal and the cursor go out the parent element vertically
is there any details in this case?

**Pull request merged**